--- origsrc/libwmf-0.2.8.4/libwmf-config.in	2004-02-10 13:41:26.000000000 +0100
+++ src/libwmf-0.2.8.4/libwmf-config.in	2012-09-05 10:40:15.750000000 +0200
@@ -2,19 +2,8 @@
 
 exec_prefix_set=no
 
-case `uname` in
-  CYGWIN*)
-    prefix=`bash -c "type -p libwmf-config"`
-    prefix=`dirname $prefix`
-    prefix=`dirname $prefix`
-    prefix=`cygpath -m "$prefix"`
-    exec_prefix="$prefix"
-  ;;
-  *)
-    prefix=@prefix@
-    exec_prefix=@exec_prefix@
-  ;;
-esac
+prefix=@prefix@
+exec_prefix=@exec_prefix@
 
 usage()
 {
@@ -96,12 +85,6 @@ fi
 wmf_cflags="@WMF_CONFIG_CFLAGS@"
 wmf_liblflags="@WMF_LIBFLAGS@"
 
-case `uname` in
-  CYGWIN*)
-    wmf_cflags=`echo $wmf_cflags | sed -e "s%@prefix@%$prefix%"`
-    wmf_liblflags=`echo $wmf_liblflags | sed -e "s%@prefix@%$prefix%"`
-  ;;
-esac
 
 libwmf_buildstyle=@LIBWMF_BUILDSTYLE@
 if test $libwmf_buildstyle = heavy; then
