--- origsrc/graphviz-2.40.1/configure.ac	2016-12-21 23:44:41.000000000 -0600
+++ src/graphviz-2.40.1/configure.ac	2017-03-06 01:30:08.314694600 -0600
@@ -91,7 +91,7 @@ AC_SUBST([GVPLUGIN_VERSION])
 GRAPHVIZ_VERSION_BUILD=0
 GRAPHVIZ_VERSION_REVISION=0
 case "${host_os}" in
-  *uwin* | *cygwin* | *mingw32* )
+  *uwin* | *mingw32* )
 	# Windows version numbers
 	# For the build number: months since Jan 2000, day of month from the timestamp
 	# For the revision number: hour, minute from the timestamp
@@ -146,10 +146,9 @@ case "${host_os}" in
 	;;
   *cygwin* )
 	CYGWIN=yes
-	DEFAULT_FONTPATH="C:/WINDOWS/FONTS;C:/WINNT/Fonts;C:/winnt/fonts"
-	PATHSEPARATOR=";"
-        NO_POSTSCRIPT_ALIAS=1
-	AC_DEFINE(NO_POSTSCRIPT_ALIAS,1,[Alias PostScript font names to URW type 1 fonts])
+	DEFAULT_FONTPATH="/cygdrive/c/Windows/Fonts:/usr/share/X11/fonts/Type1:/usr/share/X11/fonts/TTF"
+	PATHSEPARATOR=":"
+        NO_POSTSCRIPT_ALIAS=0
 	BROWSER="xdg-open"
 	JSHEXT=so
 	;;
@@ -200,7 +199,7 @@ AC_SUBST([JSHEXT])
 AC_DEFINE_UNQUOTED(DEFAULT_FONTPATH,"$DEFAULT_FONTPATH",[Path to TrueType fonts.])
 AC_DEFINE_UNQUOTED(PATHSEPARATOR,"$PATHSEPARATOR",[Path separator character.])
 AC_DEFINE_UNQUOTED(NO_POSTSCRIPT_ALIAS,$NO_POSTSCRIPT_ALIAS,[Postscript fontnames.])
-AM_CONDITIONAL(WITH_WIN32, [test "x$UWIN" = "xyes" -o "x$CYGWIN" = "xyes" -o "x$MINGW32" = "xyes"])
+AM_CONDITIONAL(WITH_WIN32, [test "x$UWIN" = "xyes" -o "x$MINGW32" = "xyes"])
 AM_CONDITIONAL(WITH_DARWIN9, [test "x$DARWIN9" = "xyes"])
 
 DEFAULT_DPI=96
@@ -254,7 +253,7 @@ if test "x$enable_shared" != "xno"; then
 	AC_ENABLE_SHARED
         use_shared="Yes"
 	case "${host_os}" in
-  		*uwin* | *cygwin* | *mingw32* )
+  		*uwin* | *mingw32* )
 			AC_DEFINE_UNQUOTED(GVDLL,1, [Define for DLLs on Windows.])
 		;;
 	esac
