--- origsrc/shared-mime-info-1.5/freedesktop.org.xml.in	2015-11-22 14:21:45.982826400 -0600
+++ src/shared-mime-info-1.5/freedesktop.org.xml.in	2015-11-22 14:16:39.124273600 -0600
@@ -2717,6 +2717,7 @@ command to generate the output files.
   </mime-type>
   <mime-type type="application/x-ms-dos-executable">
     <_comment>DOS/Windows executable</_comment>
+    <sub-class-of type="application/x-executable"/>
     <generic-icon name="application-x-executable"/>
     <magic priority="50">
       <match type="string" value="MZ" offset="0"/>
@@ -3257,6 +3258,9 @@ command to generate the output files.
       <match type="string" mask="0xffffffff000000000000000000000000ff" value="\177ELF            \003" offset="0"/>
     </magic>
     <glob pattern="*.so"/>
+    <glob pattern="*.dll"/>
+    <glob pattern="*.cmxs"/>
+    <glob pattern="*.oct"/>
   </mime-type>
   <mime-type type="application/x-shellscript">
     <_comment>shell script</_comment>
